extensions/amp-access/0.1/amp-access.js

```diff
@@ -464,16 +464,19 @@ export class AccessService {
    */
   applyAuthorizationToElement_(element, response) {
     const expr = element.getAttribute('amp-access');
-    const on = this.evaluator_.evaluate(expr, response);
-    let renderPromise = null;
-    if (on) {
-      renderPromise = this.renderTemplates_(element, response);
-    }
-    if (renderPromise) {
-      return renderPromise.then(() =>
-        this.applyAuthorizationAttrs_(element, on)
-      );
+    let on = false;
+    try {
+      on = this.evaluator_.evaluate(expr, response);
+      if (on) {
+        this.renderTemplates_(element, response).then(() =>
```

Comment:
Since `renderTemplates_` has a return type of `Promise|null`, is it the intent that this `.then` access would throw and be caught below? Also, should this be returned? Otherwise it looks like `applyAuthorizationAttrs_` could be called twice, once when this promise resolves and once after the `catch` block

Reply:
> Since renderTemplates_ has a return type of Promise|null, is it the intent that this .then access would throw and be caught below?

The old code didn't handle/care about the return of `renderTemplates_`, and I think the new version I just wrote doesn't either.

>  Also, should this be returned? 

Nice catch! Added

Reply:
What/where is the actual error that's occurring? Is it in the call to `renderTemplates_`? What caught my attention was that the returned `Promise|null` was tested in `if (renderPromise)`, whereas now it could return null and the `.then` would throw a new error

Reply:
Welp, thanks for your patience. I made a scary number of bugs in so few lines